### PR TITLE
Update coding guidelines

### PR DESCRIPTION
**Pull request by bozidarsevo**
## Description

- Add switch statements guide

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/186">17.4</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by soundsokay — COMMENTED** [review]
I'm in favor of adding this guidance. 

Maybe we could adopt the guideline from [Airbnb](https://github.com/airbnb/swift): 

> When switching over an enum, generally prefer enumerating all cases rather than using the default case.

Consider also including their code fragments which include counter examples.

**Comment by bozidarsevo** [comment]
> Maybe we could adopt the guideline from [Airbnb](https://github.com/airbnb/swift):

Thanks! I agree, their guideline looks good, should I just reference it or we can copy it and reference it with the link?

**Comment by bozidarsevo** [comment]
> should I just reference it or we can copy it and reference it with the link

what do you think @soundsokay ?

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/188">17.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: docs/switch-statements.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Switch Statements Guide
+
+## Enums
+
+When using switch statements to handle enum cases avoid handling `default` case. If we do not provide a default case and in future we add new cases, app will not build and we will immediately see where we have some API changes.
+
+Enumerating every case requires developers and reviewers have to consider the correctness of every switch statement when new cases are added in the future.
```
Comment (jaclync) [inline docs/switch-statements.md:7]: super nit:
```suggestion
Enumerating every case requires developers and reviewers to consider the correctness of every switch statement when new cases are added in the future.
```

**Review by jaclync — APPROVED** [review]
Thanks for adding the docs! 🚀

File: docs/switch-statements.md (hunk 1 above)
Comment (bozidarsevo) [reply docs/switch-statements.md:7]: had the auto merge so will tweak it separately

**Pull request merged**